src/iterate.js

```diff
@@ -49,25 +49,33 @@ class Iterate extends events.EventEmitter {
     super();
     events.EventEmitter.call(this);
 
-    assert(typeof (opts.maxIterations || 0) === 'number', 'maxIterations must be number');
-    this.maxIterations = opts.maxIterations || 0;
+    // Set default values
+    opts = _.defaults({}, opts, {
+      maxIterations: 0,
+      maxFailures: 7,
+      minIterationTime: 0,
+      waitTimeAfterFail: 0,
+    });
```

Comment:
Couldn't help myself... with this it's much easier to see what default values we have... Also less `|| 0` littering below..